### PR TITLE
Feat(hints) Implement new hints #35 and #36

### PR DESCRIPTION
**Pull request by jpcenteno**
# Feat(hints) Implement new hints #35 and #36

## Description

This PR adds the missing hints NewHint#35 and NewHint#36, now known as `BIGINT_PACK_DIV_MOD` and `BIGINT_SAFE_DIV` as requested by the issue #975.

## Checklist
- [X] Linked to Github Issue
- [X] Unit tests added
- [X] Integration tests added.
- [x] This change requires new documentation.
  - [X] Documentation has been added/updated.
  - [x] CHANGELOG has been updated.

### CODE REVIEW

File: src/hint_processor/builtin_hint_processor/bigint.rs [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<(), HintError> {
+    let res = exec_scopes.get::<BigInt>("res")?;
+    let y = exec_scopes.get::<BigInt>("y")?;
+    let x = exec_scopes.get::<BigInt>("x")?;
+    let p = exec_scopes.get::<BigInt>("p")?;
+
+    let k = safe_div_bigint(&(res * y - x), &p)?;
+    let (value, flag) = if BigInt::zero() < k {
```
Comment (Oppen) [inline src/hint_processor/builtin_hint_processor/bigint.rs:82]:
```suggestion
    let (value, flag) = if k.is_negative() {
```

**Review by Oppen — APPROVED** [review]

**Review by MegaRedHand — APPROVED** [review]

**Comment by MegaRedHand** [comment]
The wasm errors come from `std` imports. I recently fixed some in [one PR of my own](https://github.com/lambdaclass/cairo-rs/pull/1030/files/2e6cd352f2a38f040bfc68416403e1a8e8c858f3..f16eb18c94d9e339a64134b55fa9ec63f51e0e5a) (ignore the cairo files).

File: src/hint_processor/builtin_hint_processor/bigint.rs (hunk 1 above)
Comment (jpcenteno) [reply src/hint_processor/builtin_hint_processor/bigint.rs]: Addressed!

File: src/hint_processor/builtin_hint_processor/bigint.rs [hunk 2]
```diff
@@ -80,7 +80,7 @@ pub fn bigint_safe_div_hint(
     let p = exec_scopes.get::<BigInt>("p")?;
 
     let k = safe_div_bigint(&(res * y - x), &p)?;
-    let (value, flag) = if BigInt::zero() < k {
+    let (value, flag) = if k.is_negative() {
```
Comment (Oppen) [inline src/hint_processor/builtin_hint_processor/bigint.rs:83]: I saw the orders inverted when doing the suggestion. This should be `k.is_positive()`.

Comment (Oppen) [reply src/hint_processor/builtin_hint_processor/bigint.rs:83]:
```suggestion
    let (value, flag) = if k.is_positive() {
```

**Pull request merged**